### PR TITLE
[FIX] Prevent import account from auto switching accounts (5.7 Regression)

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

Video of killing/reopening app after importing two accounts, each using private key.

https://user-images.githubusercontent.com/10508597/190052314-b6ccb956-bc59-4e9b-bb16-b389c8f9c8d6.mov

Progresses [#386](https://app.zenhub.com/workspaces/mobile-release-regression-6249e5242464b70013315a98/issues/metamask/mobile-planning/386)

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by rickycodes** [comment]
Changing the runner to `macos-12` seems to have fixed this, but idk how tenable this is anymore.

It took 12m 39s :grimacing:

File: patches/@metamask+controllers+30.3.0.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/node_modules/@metamask/controllers/dist/keyring/KeyringController.js b/node_modules/@metamask/controllers/dist/keyring/KeyringController.js
+index 8b9d1ba..c67dcf6 100644
+--- a/node_modules/@metamask/controllers/dist/keyring/KeyringController.js
++++ b/node_modules/@metamask/controllers/dist/keyring/KeyringController.js
+@@ -264,7 +264,7 @@ class KeyringController extends BaseController_1.BaseController {
+      * @throws Will throw when passed an unrecognized strategy.
+      * @returns Promise resolving to current state when the import is complete.
+      */
+-    importAccountWithStrategy(strategy, args) {
++    importAccountWithStrategy(strategy, args, shouldAutoSwitchAccounts = true) {
```
Comment (rickycodes) [inline patches/@metamask+controllers+30.3.0.patch:10]: should the default be false maybe? I always thought automatically switching was weird. If it was false default you wouldn't have to pass false so much

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 name: ci
 on:
   push:
-    branches: main
```
Comment (sethkfman) [inline .github/workflows/ci.yml:4]: Why was this removed?

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by gantunesr** [comment]
@Cal-L Is there a controllers PR to address this issue without a patch for the next controllers release?
